### PR TITLE
[API] change new-api prefix to api

### PR DESCRIPTION
**Pull request by SirDomin**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | i think yes
| Deprecations?   | no
| Related tickets | 
| License         | MIT

### CODE REVIEW

**Review by GSadee — COMMENTED** [review]
There should definitely be a proper note in the UPGRADE file about these changes

File: config/packages/fos_rest.yaml [hunk 1]
```diff
@@ -7,5 +7,5 @@ fos_rest:
         empty_content: 204
     format_listener:
         rules:
-            - { path: '^/api/.*', priorities: ['json', 'xml'], fallback_format: json, prefer_extension: true }
+            - { path: '^/api/v2/.*', priorities: ['json', 'xml'], fallback_format: json, prefer_extension: true }
```
Comment (GSadee) [inline config/packages/fos_rest.yaml:10]:
```suggestion
            - { path: '^/api/v1/.*', priorities: ['json', 'xml'], fallback_format: json, prefer_extension: true }
```

File: config/packages/security.yaml [hunk 2]
```diff
@@ -1,8 +1,8 @@
 parameters:
     sylius.security.admin_regex: "^/%sylius_admin.path_name%"
-    sylius.security.api_regex: "^/api"
-    sylius.security.shop_regex: "^/(?!%sylius_admin.path_name%|new-api|api/.*|api$|media/.*)[^/]++"
-    sylius.security.new_api_route: "/new-api"
+    sylius.security.api_regex: "^/api/v1"
+    sylius.security.shop_regex: "^/(?!%sylius_admin.path_name%|api/v2/.*|api$|media/.*)[^/]++"
```
Comment (GSadee) [inline config/packages/security.yaml:4]:
```suggestion
    sylius.security.shop_regex: "^/(?!%sylius_admin.path_name%|api/.*|api$|media/.*)[^/]++"
```
?

**Review by AdamKasp — APPROVED** [review]

**Review by GSadee — APPROVED** [review]

**Review by GSadee — COMMENTED** [review]
Changes in config/packages should be reflected in UPGRADE file

File: UPGRADE-1.9.md [hunk 3]
```diff
@@ -135,3 +154,6 @@
 1. We've replaced deprecated Symfony Translator API with the new one.
    
     Replace `Symfony\Component\Translation\TranslatorInterface` with `Symfony\Contracts\Translation\TranslatorInterface` in your codebase.
+
+1. We've changed `/new-api` prefix to `/api/v2`. If you're using API, replace api request url to `/api/v2`.
```
Comment (lchrusciel) [inline UPGRADE-1.9.md:158]:
```suggestion
1. `/new-api` prefix has been changed to `/api/v2`. Please adjust your routes accordingly.
```

File: UPGRADE-1.9.md [hunk 4]
```diff
@@ -135,3 +154,6 @@
...
+   Old api is hardcoded to `/api/v1`.
```
Comment (lchrusciel) [inline UPGRADE-1.9.md:159]:
```suggestion
   Admin Api was hardcoded to `/api/v1`(instead of /api/v{version}).
```

Comment (GSadee) [reply UPGRADE-1.9.md]: If I see correctly, in Admin API there is still `/api/v{version}`

File: UPGRADE-1.9.md [hunk 5]
```diff
@@ -64,6 +64,25 @@
    +           xml_serialization:
    ```
 
+1. Unified API parameters have been changed to
+
+    ```diff
```
Comment (pamil) [inline UPGRADE-1.9.md:69]: Missing which file should I modify.

File: UPGRADE-1.9.md [hunk 6]
```diff
@@ -64,6 +64,25 @@
...
+    ```diff   
+        parameters:
+    -       sylius.security.api_regex: "^/api"
+    -       sylius.security.shop_regex: "^/(?!%sylius_admin.path_name%|new-api|api/.*|api$|media/.*)[^/]++"
+    -       sylius.security.new_api_route: "/new-api"
+    +       sylius.security.api_regex: "^/api/v1"
+    +       sylius.security.shop_regex: "^/(?!%sylius_admin.path_name%|api/.*|api$|media/.*)[^/]++"
+    +       sylius.security.new_api_route: "/api/v2"
+    ```
+1. fos_rest rules have been changed to
```
Comment (pamil) [inline UPGRADE-1.9.md:78]:
```suggestion
1. `fos_rest` rules have been changed to
```

File: UPGRADE-1.9.md [hunk 7]
```diff
@@ -64,6 +64,25 @@
...
+
+    ```diff
```
Comment (pamil) [inline UPGRADE-1.9.md:80]: Missing which file should I modify.

File: UPGRADE-1.9.md [hunk 8]
```diff
@@ -135,3 +154,6 @@
 1. We've replaced deprecated Symfony Translator API with the new one.
    
     Replace `Symfony\Component\Translation\TranslatorInterface` with `Symfony\Contracts\Translation\TranslatorInterface` in your codebase.
+
+1. `/new-api` prefix has been changed to `/api/v2`. Please adjust your routes accordingly.
+   Admin api is hardcoded to `/api/v1` (instead of `/api/v{version}`).
```
Comment (pamil) [inline UPGRADE-1.9.md:159]:
```suggestion
   Admin API is hardcoded to `/api/v1` instead of `/api/v{version}`.
```

File: tests/Controller/ProductVariantApiTest.php [hunk 9]
```diff
@@ -423,7 +423,7 @@ public function it_allows_updating_information_about_product_variant()
 <<<EOT
         {
             "code": "NEW_MUG_CODE",
-            "version": $version
+            "version": "$version"
```
Comment (pamil) [inline tests/Controller/ProductVariantApiTest.php:426]: Why is that necessary? It isn't API version.

File: UPGRADE-1.9.md (hunk 8 above)
Comment (pamil) [reply UPGRADE-1.9.md:159]: We could also add something like "Admin API will assume v1 for every request."

File: tests/Controller/ProductVariantApiTest.php (hunk 9 above)
Comment (SirDomin) [reply tests/Controller/ProductVariantApiTest.php:426]: its failing without it for some reason :/

**Review by pamil — APPROVED** [review]

**Comment by pamil** [comment]
Thanks, @SirDomin! :tada:
